memory/cloudsql/chat_message_history.go

```diff
@@ -105,3 +108,65 @@ func (c *ChatMessageHistory) validateTable(ctx context.Context) error {
 	}
 	return nil
 }
+
+// addMessage adds a new message into the ChatMessageHistory for a given
+// session.
+func (c *ChatMessageHistory) addMessage(ctx context.Context, content string, messageType llms.ChatMessageType) error {
+	data, err := json.Marshal(content)
+	if err != nil {
+		return fmt.Errorf("failed to serialize content to JSON: %w", err)
+	}
+	query := `INSERT INTO $1.$2 (session_id, data, type) VALUES ($3, $4, $5)`
+
+	_, err = c.engine.Pool.Exec(ctx, query, c.schemaName, c.tableName, c.sessionID, data, messageType)
+	if err != nil {
+		return fmt.Errorf("failed to add message to database: %w", err)
+	}
+	return nil
+}
+
+// AddMessage adds a message to the ChatMessageHistory.
+func (c *ChatMessageHistory) AddMessage(ctx context.Context, message llms.ChatMessage) error {
+	return c.addMessage(ctx, message.GetContent(), message.GetType())
+}
+
+// AddAIMessage adds an AI-generated message to the ChatMessageHistory.
+func (c *ChatMessageHistory) AddAIMessage(ctx context.Context, content string) error {
+	return c.addMessage(ctx, content, llms.ChatMessageTypeAI)
+}
+
+// AddUserMessage adds a user-generated message to the ChatMessageHistory.
+func (c *ChatMessageHistory) AddUserMessage(ctx context.Context, content string) error {
+	return c.addMessage(ctx, content, llms.ChatMessageTypeHuman)
+}
+
+// Clear removes all messages associated with a session from the
+// ChatMessageHistory.
+func (c *ChatMessageHistory) Clear(ctx context.Context) error {
+	if !c.overwrite {
+		return nil
+	}
+	query := `DELETE FROM $1.$2 WHERE session_id = $3`
+
+	_, err := c.engine.Pool.Exec(ctx, query, c.schemaName, c.tableName, c.sessionID)
```

Comment:
Does this work? I'm not sure that table name / schema can be parameterized

Reply:
I just modified the queries to make them work